### DISCUSSION TITLE
how to query PSfalcon to avoid returning duplicate hosts

### DISCUSSION THREAD

**Discussion by cesaramerico** [CrowdStrike/psfalcon#78] [Q&A] [answered]
May ask if there is some specific PSfalcon command to returning hostname/IDs based on latest 'last_seen' date?

Though it can be scripted, was looking for best approach

thanks

**Comment by bk-cs** [comment, marked as answer]
Are you trying to remove duplicates, or return results that don't have duplicates (by hostname)? 

The problem you're facing is that Falcon thinks these are unique--because devices are linked to their agent ID value--and there's no filter that can exclude them as it would require some evaluation logic by the API (which it doesn't do).

So, with that in mind, you can either use `Find-FalconDuplicate` and `Invoke-FalconHostAction` to remove the duplicates so they're not in your results:

```
$Duplicates = Find-FalconDuplicate
Invoke-FalconHostAction -Name hide_host -Ids $Duplicates.device_id
```

Or retrieve all results and filter them based on unique Hostname values:

```
$NoDupes = Get-FalconHost -Detailed -All | Sort-Object last_seen | Where-Object hostname | Group-Object hostname | ForEach-Object { $_.group | Select-Object -First 1 }
```

Since I just freehanded that second example, check it and make sure it's giving the right results in your environment. :)

**Reply by walchst** [reply]
Thanks @bk-cs. Great tip! I was having similar issues. One correction, your Get-FalconHost cmd contains a double pipe.

**Comment by cesaramerico** [comment]
Thanks all.

Em qui., 16 de set. de 2021 às 18:13, bk-cs ***@***.***>
escreveu:
